### ISSUE TITLE
Automation Schedule requiring optional start_time

### ISSUE THREAD

**Issue by darena-patrick** [hashicorp/terraform-provider-azurerm#13923]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform v1.0.4

### Affected Resource(s)

* `azurerm_automation_schedule`

### Terraform Configuration Files

```hcl
resource "azurerm_automation_schedule" "automation_schedule" {
  name                    = "aa-runbook-schedule-sql-database-backup-dev"
  resource_group_name     = azurerm_resource_group.resource_group.name
  automation_account_name = azurerm_automation_account.automation_account.name
  frequency               = "Week"
  interval                = 1
  timezone                = "UTC"
  description             = "Run this automation on a pre-determined frequency to back up SQL database."
  week_days               = ["Friday"]
}
```

### Error Output
```hcl
╷
│ Error: start_time is "2021-10-26 18:56:00 +0000 UTC" and should be at least "5m0s" in the future
│ 
│   with azurerm_automation_schedule.automation_schedule,
│   on main.tf line 471, in resource "azurerm_automation_schedule" "automation_schedule":
│  471: resource "azurerm_automation_schedule" "automation_schedule" {
│ 
╵
ERRO[0071] 1 error occurred:
        * exit status 1
```
### Expected Behaviour

According to docs, if you don't provide a start time, it should automatically set to be 7m in the future.

### Actual Behaviour

It's not doing that - so as a result I have to include start_time every time I run terraform_apply which means my `terraform plan` is going to have a change every time I run this. Previously this was not required and start_time was an optional attribute.
```hcl
start_time              = timeadd(timestamp(), "7m")
```

### Steps to Reproduce

1. `terraform apply`

### References

* https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/automation_schedule#argument-reference

**Comment by yubischiess** [comment]
I can verify that this is still a problem as of 2.84

**Comment by TaiyyabAhmed97** [comment]
Hi!, I would like to work on this issue

**Comment by atombrella** [comment]
Running one of the acceptance tests reveals some other issue with the `timezone`.

```
=== RUN   TestAccAutomationSchedule_monthly_advanced_by_day
=== PAUSE TestAccAutomationSchedule_monthly_advanced_by_day
=== CONT  TestAccAutomationSchedule_monthly_advanced_by_day
    testcase.go:110: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # azurerm_automation_schedule.test will be updated in-place
          ~ resource "azurerm_automation_schedule" "test" {
                id                      = "/subscriptions/3c388441-1a5a-4850-8d52-e39db96fa97e/resourceGroups/acctestRG-auto-211223183109356021/providers/Microsoft.Automation/automationAccounts/acctestAA-211223183109356021/schedules/acctestAS-211223183109356021"
                name                    = "acctestAS-211223183109356021"
              ~ timezone                = "Etc/UTC" -> "UTC"
                # (7 unchanged attributes hidden)
            }
```

**Comment by yubischiess** [comment]
I think Microsoft may have updated their list of timezones names ... UTC now maps to Etc/UTC. This can be silenced by changing the value from ```UTC``` to ```Etc/UTC```. This started for me a couple of months back.

**Comment by gideonmd** [comment]
This issue appears to be resolved as of azurerm v3.0.0. With the following configuration the automation is created with the start time 7 minutes in the future, as described in the documentation.

```
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "=3.0.0"
    }
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-rg"
  location = "canadacentral"
}

resource "azurerm_automation_account" "example" {
  name                = "example-account"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku_name            = "Basic"

  tags = {
    environment = "development"
  }
}

resource "azurerm_automation_schedule" "automation_schedule" {
  name                    = "aa-runbook-schedule-sql-database-backup-dev"
  resource_group_name     = azurerm_resource_group.example.name
  automation_account_name = azurerm_automation_account.example.name
  frequency               = "Week"
  interval                = 1
  timezone                = "UTC"
  description             = "Run this automation on a pre-determined frequency to back up SQL database."
  week_days               = ["Friday"]
}
```

**Comment by catriona-m** [comment]
Thanks for raising this @darena-patrick. Based on the comments it looks like this has now been resolved so I am going to close this now. However, feel free to re-open if this is not the case. Thanks!
